engine/access/apiproxy/access_api_proxy.go

```diff
@@ -47,187 +72,283 @@ func (h *FlowAccessAPIRouter) log(handler, rpc string, err error) {
+		return res, err
+	}
+
+	res, err := h.upstream.GetEventsForHeightRange(context, req)
+	h.log(UpstreamApiService, "GetEventsForHeightRange", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetEventsForBlockIDs(context context.Context, req *access.GetEventsForBlockIDsRequest) (*access.EventsResponse, error) {
-	res, err := h.Upstream.GetEventsForBlockIDs(context, req)
-	h.log("upstream", "GetEventsForBlockIDs", err)
+	if h.useIndex {
+		res, err := h.local.GetEventsForBlockIDs(context, req)
+		h.log(LocalApiService, "GetEventsForBlockIDs", err)
+		return res, err
+	}
+
+	res, err := h.upstream.GetEventsForBlockIDs(context, req)
+	h.log(UpstreamApiService, "GetEventsForBlockIDs", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetNetworkParameters(context context.Context, req *access.GetNetworkParametersRequest) (*access.GetNetworkParametersResponse, error) {
-	res, err := h.Observer.GetNetworkParameters(context, req)
-	h.log("observer", "GetNetworkParameters", err)
+	res, err := h.local.GetNetworkParameters(context, req)
+	h.log(LocalApiService, "GetNetworkParameters", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetLatestProtocolStateSnapshot(context context.Context, req *access.GetLatestProtocolStateSnapshotRequest) (*access.ProtocolStateSnapshotResponse, error) {
-	res, err := h.Observer.GetLatestProtocolStateSnapshot(context, req)
-	h.log("observer", "GetLatestProtocolStateSnapshot", err)
+	res, err := h.local.GetLatestProtocolStateSnapshot(context, req)
+	h.log(LocalApiService, "GetLatestProtocolStateSnapshot", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetProtocolStateSnapshotByBlockID(context context.Context, req *access.GetProtocolStateSnapshotByBlockIDRequest) (*access.ProtocolStateSnapshotResponse, error) {
-	res, err := h.Observer.GetProtocolStateSnapshotByBlockID(context, req)
-	h.log("observer", "GetProtocolStateSnapshotByBlockID", err)
+	res, err := h.local.GetProtocolStateSnapshotByBlockID(context, req)
+	h.log(LocalApiService, "GetProtocolStateSnapshotByBlockID", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetProtocolStateSnapshotByHeight(context context.Context, req *access.GetProtocolStateSnapshotByHeightRequest) (*access.ProtocolStateSnapshotResponse, error) {
-	res, err := h.Observer.GetProtocolStateSnapshotByHeight(context, req)
-	h.log("observer", "GetProtocolStateSnapshotByHeight", err)
+	res, err := h.local.GetProtocolStateSnapshotByHeight(context, req)
+	h.log(LocalApiService, "GetProtocolStateSnapshotByHeight", err)
 	return res, err
 }
 
 func (h *FlowAccessAPIRouter) GetExecutionResultForBlockID(context context.Context, req *access.GetExecutionResultForBlockIDRequest) (*access.ExecutionResultForBlockIDResponse, error) {
-	res, err := h.Upstream.GetExecutionResultForBlockID(context, req)
-	h.log("upstream", "GetExecutionResultForBlockID", err)
+	if h.useIndex {
+		res, err := h.local.GetExecutionResultForBlockID(context, req)
```

Comment:
were you able to confirm that this does work using local data?

Reply:
Tested it in my test and got an error, so seems like we need to upstream GetExecutionResultForBlockID.